harness/determined/profiler.py

```diff
@@ -492,7 +492,7 @@ def record_timing(
             not self.is_enabled
             or not self.timings_is_enabled
             or not self.is_active
-            or (requires_sync != self.sync_timings)
+            or (not self.sync_timings and requires_sync)
```

Comment:
This new line was a little hard for me to parse the purpose of. Maybe add a comment here indicating what this conditional is doing.

Reply:
```suggestion
            # Skip recording if this metric requires a sync to be valid and sync is disabled.
            or (not self.sync_timings and requires_sync)
```